### PR TITLE
GFORMS-1831: Add support so that headers can be included in miniSummaryList

### PR DESCRIPTION
**Pull request by lxol**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala [hunk 1]
```diff
@@ -1115,54 +1116,73 @@ class SectionRenderingService(
...
+                  validationResult,
+                  ei.envelope,
+                  ei.addressRecordLookup,
+                  None,
+                  FastForward.StopAt(ei.sectionNumber)
+                )
+            }
+            .toList
+            .flatten
+        case MiniSummaryRow.HeaderRow(header) => ???
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala:1163]: Nope

File: app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala [hunk 2]
```diff
@@ -1115,54 +1116,73 @@ class SectionRenderingService(
...
+          case _: MiniSummaryRow.HeaderRow => List(a) :: b
+          case _: MiniSummaryRow.ValueRow  => (a :: b.head) :: b.tail
+        }
+      )
+      .reverse
+      .map(_.reverse)
+    val htmls = visibleRowsPartitioned.map {
+      case MiniSummaryRow.HeaderRow(h) :: xs =>
+        HtmlFormat.fill(List((new header)(Html(sse(h, false))), aux(xs)))
+      case xs => aux(xs)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala:1181]: Can we use `loop` instead?

**Review by josef-vlach — CHANGES_REQUESTED** [review]
XXX

File: app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala (hunk 1 above)
Comment (lxol) [reply app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala:1163]: oops

Comment (lxol) [reply app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala:1163]: I will throw Exception here as `aux` method does't expect `HeaderRow` in the list of MiniSummaryRow

File: app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala (hunk 2 above)
Comment (lxol) [reply app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala]: renamed

File: app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala (hunk 1 above)
Comment (lxol) [reply app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala]: done

**Review by josef-vlach — CHANGES_REQUESTED** [review]

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**